community/cypher/cypher-planner-3.4/src/main/scala/org/neo4j/cypher/internal/compiler/v3_4/planner/logical/steps/LogicalPlanProducer.scala

```diff
@@ -36,7 +37,7 @@ import org.neo4j.cypher.internal.v3_4.logical.plans.{DeleteExpression => DeleteE
  * No other functionality or logic should live here - this is supposed to be a very simple class that does not need
  * much testing
  */
-case class LogicalPlanProducer(cardinalityModel: CardinalityModel, readTransactionLayer: Int, val idGen : IdGen) extends ListSupport {
+case class LogicalPlanProducer(cardinalityModel: CardinalityModel, readTransactionLayer: Int, readtxLayerAttribute: TransactionLayerAttribute, idGen : IdGen) extends ListSupport {
```

Comment:
How about changing 
 * `readTransactionLayer` -> `currentTxLayer`
 * `readtxLayerAttribute` -> `txLayers`
 * `TransactionLayerAttribute` -> `TransactionLayer`

Reply:
I agree to shortening and removing the "Attribute" from the names. But I do want to keep "read" in the val name. It makes it very clear that this is the layer an operator reads from, but not the one it writes to (if it is an operator that writes).